src/callbacks_step/save_restart_dg.jl

```diff
@@ -157,7 +157,12 @@ function load_restart_file(mesh::Union{ParallelTreeMesh, ParallelP4estMesh}, equ
   if !mpi_isroot()
     # Receive nodal data from root
     for v in eachvariable(equations)
-      MPI.Scatterv!(nothing, @view(u[v, .., :]), mpi_root(), mpi_comm())
+      if length(u[v, .., :]) == 0
+        data = eltype(u[v, .., :])[]
+      else
+        data = @view u[v, .., :]
```

Comment:
Please avoid copying the data unnecessarily (slices are copies in Julia):
```suggestion
      if isempty(u)
        data = eltype(u)[]
      else
        data = @view u[v, .., :]
```
This will introduce a type instability here. I don't know how severe it is but it may be fixed by copying the scatter command to each block.